synapse/storage/databases/main/end_to_end_keys.py

```diff
@@ -755,81 +755,135 @@ async def claim_e2e_one_time_keys(
         """
 
         @trace
-        def _claim_e2e_one_time_keys(txn):
-            sql = (
-                "SELECT key_id, key_json FROM e2e_one_time_keys_json"
-                " WHERE user_id = ? AND device_id = ? AND algorithm = ?"
-                " LIMIT 1"
+        def _claim_e2e_one_time_key_simple(
+            txn, user_id: str, device_id: str, algorithm: str
+        ) -> Optional[Tuple[str, str]]:
+            """Claim OTK for device for DBs that don't support RETURNING."""
```

Comment:
probably worth documenting what this, and `_claim_e2e_one_time_key_returning` returns